public/app/plugins/gen.go

```diff
@@ -54,52 +53,52 @@ func main() {
 	})
 
 	outputFile := filepath.Join("pkg", "plugins", "pfs", "corelist", "corelist_load_gen.go")
-	pluginTreeListJenny := codegen.PluginTreeListJenny(outputFile)
-	pluginKindGen.AppendManyToOne(pluginTreeListJenny)
+	pluginKindGen.Append(codegen.PluginTreeListJenny(outputFile),
+		codejen.AdaptOneToMany(codegen.FlattenJenny(corecodegen.GoTypesJenny("pkg/tsdb", nil)), mapToDecls))
```

Comment:
@sdboyer me and @mckn have explored two different ways of implementing adapters for converting `*codegen.PluginTreeDeclForGen` to `[]*corecodegen.DeclForGen`. This is one way

Reply:
I'm not quite sure what the key difference you're pointing to is here, given that `mapToDecls` is a stub function right now. But this appears to be more what i'd recommend - using the `codejen` adapter builder and passing it a function that does the conversion on the fly.

Reply:
Sorry, I added another comment for the other way below, but here it goes in fully:
https://github.com/grafana/grafana/blob/5f516f14986dc46b9c6c4cbbe55688cf32d7ac7c/public/app/plugins/gen.go#L66-L104

Reply:
Ahh! ok, then it's this approach. There's no need for `FlattenJenny`.

To reuse `GoTypesJenny` and `TSTypesJenny`, you really shouldn't have to make it all the way back to a `DeclForGen`. I've got a PR i'll put up in a bit that refactors them to take a much narrower object that should be easier for you to write a mapping func for.

Reply:
Yeah, I also think we should change the level of the object we work with to a lower level. Currently we pass a pfs.Tree which can contains multiple Linages. So I think flattening those to a level where we have one Linage should probably help as well.

Reply:
Yep!

I think something like the meta-jenny thing i'm doing over in #58995 is probably what you want.